### PR TITLE
XMLHttpRequest getAllResponseHeaders()

### PR DESCRIPTION
**Pull request by annevk**
See https://github.com/whatwg/xhr/issues/108 for context.

### CODE REVIEW

**Comment by tobie** [comment]
Ping @wpt-pr-bot.

File: XMLHttpRequest/getallresponseheaders.htm [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<!doctype html>
+<title>XMLHttpRequest: getAllResponseHeaders()</title>
```
Comment (alvaropratama) [inline XMLHttpRequest/getallresponseheaders.htm]: respone

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/208629405)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 999d8502b91a468336cf82529dc6c0ce67d7f15a
Using browser at version BuildID 20170306110339; SourceStamp 966464a68a2cb3ca1125808e34abb5c1d34e3797
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/4641/XMLHttpRequest/getallresponseheaders.htm">/XMLHttpRequest/getallresponseheaders.htm</a></summary>


|                  Subtest                  | Results |                                                                      Messages                                                                     |
|-------------------------------------------|---------|---------------------------------------------------------------------------------------------------------------------------------------------------|
|                                           | OK      |                                                                                                                                                   |
| `XMLHttpRequest: getAllResponseHeaders()` | FAIL    | `assert_equals: expected "also-here: Mr. PB\r\newok: lego\r\nfoo-test: 1, 2\r\n" but got "foo-TEST: 1, 2\r\nALSO-here: Mr. PB\r\newok: lego\r\n"` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/208629406)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 999d8502b91a468336cf82529dc6c0ce67d7f15a
Using browser at version 58.0.3026.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/4641/XMLHttpRequest/getallresponseheaders.htm">/XMLHttpRequest/getallresponseheaders.htm</a></summary>


|                  Subtest                  | Results |                                                                      Messages                                                                     |
|-------------------------------------------|---------|---------------------------------------------------------------------------------------------------------------------------------------------------|
|                                           | OK      |                                                                                                                                                   |
| `XMLHttpRequest: getAllResponseHeaders()` | FAIL    | `assert_equals: expected "also-here: Mr. PB\r\newok: lego\r\nfoo-test: 1, 2\r\n" but got "ALSO-here: Mr. PB\r\nfoo-TEST: 1, 2\r\newok: lego\r\n"` |

</details>
</details>

**Comment by domenic** [comment]
So the failures are expected because we haven't solved header casing yet? And this is mainly testing the comma-spacing? If so, LGTM.

**Comment by annevk** [comment]
My plan is to file implementation bugs to see if other implementers can align with WebKit for get(All)ResponseHeader(s) (at least Firefox did not want to for setRequestHeader and nobody else had either, so I'm reverting that). If they object, we'll have to make it more complicated and expose sorting/casing.

**Comment by domenic** [comment]
Hmm OK. It might be good to have a separate test to only test the comma-separatedness then? Just so as to not test two things at once.

**Comment by annevk** [comment]
Yeah, XMLHttpRequest/getresponseheader-case-insensitive.htm does so already in a line that is not modified here. Turns out we never updated the tests to begin with to remove the spaces there...

**Comment by domenic** [comment]
Sure, but it might be good to do one for getAllResponseHeaders() as well.

(The test in this PR looks good for the longer-term plan; I'm just suggesting additional coverage for the shorter-term goal of getting everyone to use comma-space.)

**Comment by annevk** [comment]
Added a test for that.

**Review by domenic — APPROVED** [review]

**Pull request merged**